### ISSUE TITLE
Task Incremental Scores?

### ISSUE THREAD

**Issue by Susmit-A** [tdemin16/Continual-LayerNorm-Tuning#2]
Hi,

I enjoyed going through your work. I would appreciate it if you could provide your model's scores in a TIL setting or let me know how to get them using your code.

Thanks,
Susmit

**Comment by tdemin16** [comment]
Hi, thank you for taking an interest in our work. Unfortunately, we did not compute the TIL scores. However, with little code, you should be able to compute them.
In practice, in TIL, the task id is known at inference time, so you could provide the vector of IDs as input to the ViT forward by passing it to the `task_selection` argument and leaving the `original features=None`. Then, you should also mask the outputs. To do so, you should call a modified version of `utils.mask_logits(...)` (that you must implement), where given a vector of task IDs, it outputs a mask for each sample.

I hope this can help you with your problem.
Thomas

**Comment by Snimm** [comment]
Hi,
Regarding the `WeightSelectionMechanism` class, I've noticed that changing the `forward` method from `sim, tasks = torch.max(sim, dim=-1)` to `tasks = torch.full((B, ), task_eval)` yields the same accuracy as the oracle reported in your paper. 
Could you please clarify if this modification is valid for getting task incremental scores?

**Comment by tdemin16** [comment]
Yes, you are correct. However you  must also mask output logits to retrieve only those for the correct task. If you do both things you should be ok.
